lightning/src/ln/outbound_payment.rs

```diff
@@ -282,6 +282,40 @@ impl PendingOutboundPayment {
 		insert_res
 	}
 
+	pub(super) fn insert_from_monitor_on_startup(
+		&mut self, session_priv: [u8; 32], path: &Path, payment_hash: PaymentHash,
+		best_block_height: u32
+	) -> bool {
+		match self {
+			PendingOutboundPayment::AwaitingInvoice { .. } |
+				PendingOutboundPayment::InvoiceReceived { .. } => {
+					// If we've reached this point, it means we initiated a payment to a BOLT 12 invoice and
+					// locked the htlc(s) into the `ChannelMonitor`(s), but failed to persist the
+					// `ChannelManager` after transitioning from this state to `Retryable` prior to shutdown.
+					// Therefore, we need to move this payment to `Retryable` now to avoid double-paying if
+					// the recipient sends a duplicate invoice.
+					*self = Self::Retryable {
+						retry_strategy: None,
+						attempts: PaymentAttempts::new(),
+						payment_params: None,
+						session_privs: hash_set_from_iter([session_priv]),
+						payment_hash,
+						payment_secret: None,
+						payment_metadata: None,
+						keysend_preimage: None,
+						custom_tlvs: Vec::new(),
+						pending_amt_msat: path.final_value_msat(),
+						pending_fee_msat: Some(path.fee_msat()),
+						total_msat: path.final_value_msat(),
+						starting_block_height: best_block_height,
+						remaining_max_total_routing_fee_msat: None,
```

Comment:
Is it a problem that some of these are `None` / different from the values previously set? Perhaps we should comment on any ramifications.

Reply:
Let me know if the new clarification comment helps!